### PR TITLE
Discovery dataset event stream

### PR DESCRIPTION
**Pull request by LtChae**
This will necessitate a migration, which will likely be a part of the discovery_api code base until the registry is completely purged in a tech debt cleanup.

We elected to use the existing Model struct as the view state structure, isolating the scope refactor as much as possible. The bulk of the changes are to tests.

### CODE REVIEW

File: lib/discovery_api/data/model.ex [hunk 1]
```diff
@@ -45,22 +46,28 @@ defmodule DiscoveryApi.Data.Model do
...
-    |> struct_from_json()
+    {:ok, model} = Brook.ViewState.get(DiscoveryApi.instance(), :models, id)
+
+    model
     |> add_system_attributes()
   end
 
   def get_all() do
-    get_all_models()
+    {:ok, models} = Brook.ViewState.get_all(DiscoveryApi.instance(), :models)
```
Comment (mcsearchin) [inline lib/discovery_api/data/model.ex]: Is there a `get_all!` function that we should use here instead?

Comment (LtChae) [reply lib/discovery_api/data/model.ex]: We were surprised yesterday to learn the answer was no.

File: lib/discovery_api/data/registry_migrator.ex [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def put_dataset(dataset, env) do
+    IO.puts("Posting dataset #{dataset.id}")
+
+    case put("https://andi.#{env}.internal.smartcolumbusos.com/api/v1/dataset", Jason.encode!(dataset),
+           headers: [{"content-type", "application/json"}]
+         ) do
+      {:ok, result} -> result
+      error -> IO.puts("Got error posting dataset #{dataset.id}: #{inspect(error)}")
```
Comment (mcsearchin) [inline lib/discovery_api/data/registry_migrator.ex]: I'm guessing this file is only temporary, but should we use `Logger` instead of `IO.puts`?

Comment (LtChae) [reply lib/discovery_api/data/registry_migrator.ex]: Perhaps, but I don't believe it to be worth a half hour build to fix.

File: lib/discovery_api/event_handler.ex [hunk 3]
```diff
@@ -20,4 +22,29 @@ defmodule DiscoveryApi.EventHandler do
...
+
+  def handle_event(%Brook.Event{type: dataset_update(), data: %Dataset{} = dataset}) do
+    Logger.debug(fn -> "Handling dataset: `#{dataset.technical.systemName}`" end)
+
+    with {:ok, organization} <- DiscoveryApi.Schemas.Organizations.get_organization(dataset.technical.orgId),
+         {:ok, _cached} <- SystemNameCache.put(dataset, organization),
+         model <- Mapper.to_data_model(dataset, organization) do
+      DiscoveryApi.Search.Storage.index(model)
+      save_dataset_to_recommendation_engine(dataset)
+      ResponseCache.invalidate()
```
Comment (mcsearchin) [inline lib/discovery_api/event_handler.ex]: Does the order of these actions after the `do` matter?  And if they fail, will they raise or just fail silently?

File: lib/discovery_api/data/registry_migrator.ex (hunk 2 above)
Comment (mcsearchin) [reply lib/discovery_api/data/registry_migrator.ex]: Totes fair.

File: lib/discovery_api/data/model.ex (hunk 1 above)
Comment (mcsearchin) [reply lib/discovery_api/data/model.ex]: Bummer.

File: lib/discovery_api/event_handler.ex (hunk 3 above)
Comment (mcsearchin) [reply lib/discovery_api/event_handler.ex]: After discussion - sounds like this was just lifted from the registry subscriber event listener.  All clear.

File: test/unit/discovery_api/data/dataset_update_event_handler_test.exs [hunk 4]
```diff
@@ -1,85 +1,82 @@
...
       dataset = TDG.create_dataset(dataset_map)
-      organization = TDG.create_schema_organization(%{id: dataset.technical.orgId})
+      organization = create_schema_organization(%{id: dataset.technical.orgId})
       allow(Organizations.get_organization(dataset.technical.orgId), return: {:ok, organization})
-      allow(Model.save(any()), return: {:ok, :success})
 
-      DatasetEventListener.handle_dataset(dataset)
+      Brook.Test.send(DiscoveryApi.instance(), dataset_update(), "unit", dataset)
 
       assert called == called?(DiscoveryApi.RecommendationEngine.save(dataset))
```
Comment (ManApart) [inline test/unit/discovery_api/data/dataset_update_event_handler_test.exs]: Looks like this should be `assert_called` not `assert called`

**Pull request merged**